plugins/repository-s3/src/main/java/org/elasticsearch/cloud/aws/InternalAwsS3Service.java

```diff
@@ -138,8 +99,8 @@ private synchronized AmazonS3 getClient(String endpoint, String protocol, String
         }
 
         // #155: we might have 3rd party users using older S3 API version
-        String awsSigner = settings.get(CLOUD_S3.SIGNER, settings.get(CLOUD_AWS.SIGNER));
-        if (awsSigner != null) {
+        String awsSigner = CLOUD_S3.SIGNER_SETTING.get(settings);
+        if (Strings.hasText(awsSigner)) {
             logger.debug("using AWS API signer [{}]", awsSigner);
             AwsSigner.configureSigner(awsSigner, clientConfiguration, endpoint);
```

Comment:
Totally exact. I changed the behavior for ec2. We now throw the `IllegalArgumentException` everywhere.